### ISSUE TITLE
ERROR The type 'ValueType' is defined in an assembly that is not referenced. You must add a reference to assembly 'netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'

### ISSUE THREAD

**Issue by idseefeld** [umbraco/Umbraco-CMS#10700]
### Which *exact* Umbraco version are you using? For example: 8.13.1 - don't just write v8

8.15.0

### Bug summary

Calling extension method .FirstOrDefault on an IEnumerable<T> object throws an error with description:

_The type 'ValueType' is defined in an assembly that is not referenced. You must add a reference to assembly 'netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'_

```
var roots = Umbraco.ContentAtRoot();
var root = roots.FirstOrDefault();
```

I found a solution on stakeoverflow: https://stackoverflow.com/questions/49925484/you-must-add-a-reference-to-assembly-netstandard-version-2-0-0-0

Essentially you have to add this to the Web.config:

```
<system.web>
...
  <compilation defaultLanguage="c#" debug="true" batch="true" targetFramework="4.7.2" numRecompilesBeforeAppRestart="50">
    <assemblies>
      <add assembly="netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51"/>
    </assemblies>
  </compilation>
...
```
by replacing:

```
<system.web>
...
<compilation defaultLanguage="c#" debug="true" batch="true" targetFramework="4.7.2" numRecompilesBeforeAppRestart="50" />
...

```

### Specifics

The issue has been added between v8.15.0-rc and final v8.15.0.
I could not figure out which change introduced the issue.
Maybe the use of System.Collections.Immutable as this was added after _RC_ release.

### Steps to reproduce

In a fresh v8.15.0 install create e.g. a simple WebApi class in `\App_Code\SomeController.cs` folder:

```
public class SomeController : UmbracoApiController
{
  public IEnumerable<int> GetIds()
  {
    var roots = Umbraco.ContentAtRoot();
    var root = roots.FirstOrDefault(); //here you get the error
    return null;
  }
}
```

**Comment by idseefeld** [comment]
Don't get confused by building without errors. But IntelliSens will show you the issue and runtime error message.

**Comment by emma-hq** [comment]
Hiya Dirk!

Thanks for reporting this and adding a PR for it. I'll take a look and let you know if I need anything right here :)

Emma

**Comment by emma-hq** [comment]
Hey Dirk,

I've had a look and actually, we're already in the middle of a fix for this one so we'll be merging that in and will report back here to close the issue when it is done. 

Emma

**Comment by idseefeld** [comment]
Thank you @emma-hq!

Dirk

**Comment by nul800sebastiaan** [comment]
This should be fixed with https://github.com/umbraco/Umbraco-CMS/pull/10772

**Comment by Bellezze** [comment]
Has this issue been resolved in 8.15.2?

**Comment by idseefeld** [comment]
This will be solved in version 8.16.0.
